ci/tests/arakoon/extend_promote_demote_test.py

```diff
@@ -122,108 +143,74 @@ def teardown():
             cmd = 'rm -rf {0}'.format(location)
             general.execute_command_on_node(ip, cmd)
 
+    for key in KEY_CLEANUP:
+        cmd = 'etcdctl rm /ovs/arakoon/{0} --recursive'.format(key)
+        general.execute_command(cmd)
+
 
 def is_arakoon_dir_config_structure_cleaned_up(ip, cluster_name, base_dir='/mnt/storage'):
-    config_dir = ArakoonClusterConfig.ARAKOON_CONFIG_DIR.format(cluster_name)
-    config_file = ArakoonClusterConfig.ARAKOON_CONFIG_FILE.format(cluster_name)
+    config_file = ArakoonClusterConfig.ETCD_CONFIG_KEY.format(cluster_name)
     tlog_dir = ArakoonInstaller.ARAKOON_TLOG_DIR.format(base_dir, cluster_name)
     log_dir = ArakoonInstaller.ARAKOON_LOG_DIR.format(cluster_name)
     home_dir = ArakoonInstaller.ARAKOON_HOME_DIR.format(base_dir, cluster_name)
 
     client = SSHClient(ip, username='root')
-    assert_false(client.file_exists(config_file)),\
-        "Arakoon config file {0} still exists on {1}".format(config_file, ip)
-    for directory in [config_dir, tlog_dir, home_dir, log_dir]:
+    assert_false(config_key_exists_on_node(ip, config_file)),\
+        "Arakoon configuration in etcdctl still exists on {0}".format(ip)
+    for directory in [tlog_dir, home_dir, log_dir]:
         assert_false(client.dir_exists(directory)),\
-            "Arakoon directory {0} still exists on {1}".format(config_file, ip)
+            "Arakoon directory {0} still exists on {1}".format(directory, ip)
+
+
+def config_key_exists_on_node(ip, config_file):
+    out = general.execute_command_on_node(ip, 'etcdctl ls {0} --recursive'.format(BASE_KEY_DIR))
```

Comment:
The etcd extensions has an EtcdConfiguration.exists() which is much easier and cleaner